packages/did/src/DidLinks/AccountLinks.chain.ts

```diff
@@ -247,154 +170,51 @@ export async function queryConnectedAccountsForDid(
 export async function queryWeb3Name(
   linkedAccount: Address
 ): Promise<Web3Name | null> {
+  const api = ConfigService.get('api')
   // TODO: Replace with custom RPC call when available.
-  const linkedDid = await queryConnectedDidForAccount(linkedAccount)
-  if (!linkedDid) {
+  const encoded = await api.query.didLookup.connectedDids(
+    accountToChain(linkedAccount)
+  )
+  if (encoded.isNone) {
     return null
   }
-  return queryWeb3NameForDid(linkedDid)
-}
-
-/**
- * Return true whether the provided account has been linked to the provided DID.
- *
- * @param did The DID to use for the lookup.
- * @param account The account to use for the lookup.
- * @returns True if the DID and account is linked, false otherwise.
- */
-export async function queryIsConnected(
-  did: DidUri,
-  account: Address
-): Promise<boolean> {
-  const api = ConfigService.get('api')
-  if (isEthereumEnabled(api)) {
-    // The following function returns something different than 0x00 if there is an entry for the provided key, 0x00 otherwise.
-    return !(
-      await api.query.didLookup.connectedAccounts.hash(
-        encodeDid(did),
-        encodeMultiAddress(account)
-      )
-    ).isEmpty
-    // isEmpty returns true if there is no entry for the given key -> the function should return false.
-  }
-  return !(
-    await api.query.didLookup.connectedAccounts.hash(encodeDid(did), account)
-  ).isEmpty
-}
-
-/**
- * Retrieves the deposit amount to link an account to a DID as currently stored in the runtime.
- *
- * @returns The deposit amount. The value is indicated in femto KILTs.
- */
-export async function queryDepositAmount(): Promise<BN> {
-  const api = ConfigService.get('api')
-  return api.consts.didLookup.deposit.toBn()
+  const { did } = connectedDidFromChain(encoded)
+  return web3NameFromChain(await api.query.web3Names.names(didToChain(did)))
```

Comment:
let's not decode and then re-encode, encoded.did should work as a parameter

Reply:
Done in https://github.com/KILTprotocol/sdk-js/pull/613/commits/bfc1d384d7762a71f4437ae9d00a1be94446b11a